File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.java
public class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int n = students.length, circleStudentCount = 0, squareStudentCount = 0;
        // count how many students prefer each sandwich type
        for(var i=0; i<n; i++){
            if(students[i] == 0) circleStudentCount++;
            else squareStudentCount++;
        }

        // for every sandwich, if there is a student anywhere who can eat, then continue, else break
        var k = 0;
        for(; k < n; k++){
            if(sandwiches[k] == 0){
                if(circleStudentCount == 0) break;
                else circleStudentCount--;
            }
            else{
                if(squareStudentCount == 0) break;
                else squareStudentCount--;
            }
        }

        // k students could eat sandwiches
        // so those who could not eat is total minus k
        return n-k;
    }
}

File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        n = len(students)

        # Count how many students prefer each sandwich type
        circle_student_count = sum(1 for stud in students if stud == 0)
        square_student_count = n - circle_student_count
        
        # For every sandwich, if there is a student anywhere who can eat, then continue, else break
        k = 0
        for k in range(n):
            if sandwiches[k] == 0:
                if circle_student_count == 0:
                    break
                circle_student_count -= 1
            else:
                if square_student_count == 0:
                    break
                square_student_count -= 1
        else:
            k = n   # when loop ends, then set k to n

        # k students could eat sandwiches
        # So those who could not eat is total minus k
        return n-k